be/src/storage/compaction_scheduler.cpp

```diff
@@ -90,15 +90,8 @@ bool CompactionScheduler::_can_do_compaction_task(Tablet* tablet, CompactionTask
     // to compatible with old compaction framework
     // TODO: can be optimized to use just one lock
     int64_t last_failure_ms = 0;
-    uint16_t task_num = StorageEngine::instance()->compaction_manager()->running_tasks_num_for_type(
-            compaction_task->compaction_type());
     DataDir* data_dir = tablet->data_dir();
     if (compaction_task->compaction_type() == CUMULATIVE_COMPACTION) {
-        if (config::max_cumulative_compaction_task >= 0 && task_num >= config::max_cumulative_compaction_task) {
```

Comment:
why remove this check?

Reply:
The function of `config::max_cumulative_compaction_task)` is somewhat duplicated by the function of `config::cumulative_compaction_num_threads_per_disk`, lator we will check task_num less than `config::cumulative_compaction_num_threads_per_disk`. 
Same reason for `config::base_compaction_num_threads_per_disk`.

On the other hand, we already have the `max_compaction_task_num` to make sure the overall task_num will not exceed the limit, i think there is no need to set another maximum task limit for each type of compaction.

Reply:
@chaoyli should double-check this